sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java

```diff
@@ -56,6 +58,57 @@
         this.encryptedBlobAsyncClient = encryptedBlobAsyncClient;
     }
 
+    /**
+     * Creates and opens an output stream to write data to the block blob.
+     *
+     * @return A {@link BlobOutputStream} object used to write data to the blob.
+     * @throws BlobStorageException If a storage service error occurred.
+     */
+    public BlobOutputStream getBlobOutputStream() {
+        return getBlobOutputStream(false);
+    }
+
+    /**
+     * Creates and opens an output stream to write data to the block blob.
+     *
+     * @return A {@link BlobOutputStream} object used to write data to the blob.
+     * @param overwrite Whether or not to overwrite, should data exist on the blob.
+     * @throws BlobStorageException If a storage service error occurred.
+     */
+    public BlobOutputStream getBlobOutputStream(boolean overwrite) {
+        BlobRequestConditions requestConditions = null;
+        if (!overwrite) {
+            if (exists()) {
+                throw logger.logExceptionAsError(new IllegalArgumentException(Constants.BLOB_ALREADY_EXISTS));
+            }
```

Comment:
This case is interesting given that writing to the blob may be delayed a non-determinant amount of time, this differs from the other cases of `overwrite` where their write operation should happen relatively soon compared to the validation check.

Reply:
This makes sense to me, the only other way would be to pass it in to the outputstream and it would fail upon closing - so that would be potentially annoying for the customer

Reply:
Agreed, failing fast is the better option here.